File: Paper-WolfLinkRPC/src/main/java/org/wolflink/paper/wolflinkrpc/command/RemoteAPICall.java
package org.wolflink.paper.wolflinkrpc.command;

import org.jetbrains.annotations.NotNull;
import org.wolflink.common.wolflinkrpc.api.annotations.CommandFunction;
import org.wolflink.common.wolflinkrpc.api.interfaces.CallbackFunction;
import org.wolflink.common.wolflinkrpc.entity.RPCDataPack;
import org.wolflink.common.wolflinkrpc.entity.impl.RemoteAPICallImpl;
import org.wolflink.paper.wolflinkrpc.App;

@CommandFunction
public class RemoteAPICall extends RemoteAPICallImpl {
    public RemoteAPICall() {
        super(new CallbackFunction() {
            @Override
            public void success(@NotNull RPCDataPack datapack) {
                String info = datapack.getJsonObject().get("info").getAsString();
                App.RPC_CONFIGURATION.getLogger().info(info);
            }

            @Override
            public void failed(@NotNull RPCDataPack datapack) {
                CallbackFunction.DefaultImpls.failed(this,datapack);
            }
        });
    }
}

File: Windows-WolfLinkRPC/src/main/kotlin/command/RemoteAPICall.kt
package org.wolflink.windows.wolflinkrpc.command

import org.wolflink.common.wolflinkrpc.api.annotations.CommandFunction
import org.wolflink.common.wolflinkrpc.api.interfaces.CallbackFunction
import org.wolflink.common.wolflinkrpc.entity.RPCDataPack
import org.wolflink.common.wolflinkrpc.entity.impl.command.RemoteAPICallImpl
import org.wolflink.windows.wolflinkrpc.RPCLogger

@CommandFunction
class RemoteAPICall : RemoteAPICallImpl(object : CallbackFunction{
    override fun success(datapack: RPCDataPack) {
        RPCLogger.info(datapack.jsonObject.get("info").asString)
    }
    override fun failed(datapack: RPCDataPack) {
        RPCLogger.info("响应超时，未能在指定时间内接收到来自客户端的回应。")
    }
})